src/types/contract.rs

```diff
@@ -1,42 +1,160 @@
+pub use build_info::BuildInfo;
 use serde::{Deserialize, Serialize};
 
 /// The struct provides information about deployed contract's source code and supported standards.
 ///
-/// Contract source metadata follows [NEP-330 standard](https://github.com/near/NEPs/blob/master/neps/nep-0330.md) for smart contract verification
+/// Contract source metadata follows [**NEP-330 standard**](https://github.com/near/NEPs/blob/master/neps/nep-0330.md) for smart contracts
+///
+/// See documentation of [`crate::Contract::contract_source_metadata`] on how to query this for a contract via this crate
+// `rustdoc` clearly lacks functionality of automatic backlinks within a single crate
 #[derive(Debug, Clone, PartialEq, Default, Eq, Serialize, Deserialize)]
 pub struct ContractSourceMetadata {
-    /// Optional version identifier, typically a commit hash or semantic version
+    /// Optional version identifier, typically a semantic version
+    ///
+    /// ## Examples:
     ///
-    /// Examples:
-    /// - Git commit: "39f2d2646f2f60e18ab53337501370dc02a5661c"
-    /// - Semantic version: "1.0.0"
+    /// ```rust,no_run
+    /// # let version: Option<String> =
+    /// // Semantic version
+    /// Some("1.0.0".into())
+    /// # ;
+    /// ```
     pub version: Option<String>,
 
     // cSpell::ignore bafybeiemxf5abjwjbikoz4mc3a3dla6ual3jsgpdr4cjr3oz3evfyavhwq
-    /// Optional URL to source code repository or IPFS CID
+    /// Optional URL to source code repository/tree
+    ///
+    /// ## Examples:
     ///
-    /// Examples:
-    /// - GitHub URL: "<https://github.com/near-examples/nft-tutorial>"
-    /// - IPFS CID: "bafybeiemxf5abjwjbikoz4mc3a3dla6ual3jsgpdr4cjr3oz3evfyavhwq"
```

Comment:
same

Reply:
according to comment

https://github.com/near/near-sdk-rs/pull/1249#pullrequestreview-2453152605

i planned that on one beautiful day i would make value from env var `NEP_330_LINK` take precedence
over value from  `#[near(contract_metadata(...)]`  .

Thus, according to logic in 

https://github.com/near/cargo-near/blob/main/cargo-near-build/src/types/near/build/buildtime_env/link.rs?plain=1#L10
,
when building with `cargo-near`, 

1. value from dockerbuild will have most priority 
2. then value from crate.metadata.repository will take effect, if it's not a docker build
3. if there's no `package.repository` set for contract's crate `#[near(contract_metadata(...)]` will come out as it will be no longer shadowed.